src/tty/unix.rs

```diff
@@ -335,26 +366,47 @@ impl EventedReadWrite for Pty {
     ) -> io::Result<()> {
         self.token = (*token.next().unwrap()).into();
         poll.register(
-            &EventedFd(&self.raw_fd),
+            &EventedFd(&self.fd.as_raw_fd()),
             self.token,
             interest,
             poll_opts
+        )?;
+
+        self.sig_token = (*token.next().unwrap()).into();
+        poll.register(
+            &EventedFd(&self.sig_pipe_read.as_raw_fd()),
+            self.sig_token,
+            mio::Ready::readable(),
+            mio::PollOpt::level()
         )
     }
```

Comment:
Sticking the `register`/`reregister`/`deregister` together like this seems really gross, but I'm hard-pushed to think up a better approach.